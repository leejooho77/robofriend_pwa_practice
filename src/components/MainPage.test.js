import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';

let wrapper;
let wrapper2;
let wrapper3;

beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }
    wrapper = shallow(<MainPage { ...mockProps }/>);
})

describe('Render MainPage', () => {
    test('Expect to render MainPage', () => {
        expect(wrapper).toMatchSnapshot();
    })
})

describe('Testing FilterRobot function', () => {
    beforeEach(() => {
        const mockProps2 = {
            onRequestRobots: jest.fn(),
            robots: [
                {
                    id: 3,
                    name: 'John',
                    email: 'john@gmail.com'
                }
            ],
            searchField: 'John',
            isPending: false
        }
        wrapper2 = shallow(<MainPage { ...mockProps2 } />);

        const mockProps3 = {
            onRequestRobots: jest.fn(),
            robots: [
                {
                    id: 3,
                    name: 'John',
                    email: 'john@gmail.com'
                }
            ],
            searchField: 'a',
            isPending: false
        }
        wrapper3 = shallow(<MainPage { ...mockProps3 } />);
    })

    test('FilterRobot should return empty array', () => {
        expect(wrapper.instance().filterRobots()).toEqual([]);
    })
    
    test('FilterRobot should return array with John', () => {
        expect(wrapper2.instance().filterRobots()).toEqual([
            {
                id: 3,
                name: 'John',
                email: 'john@gmail.com'
            }
        ]);
    })
    
    test('FilterRobot should return empty array when there is no match', () => {
        expect(wrapper3.instance().filterRobots()).toEqual([]);
    })
})
