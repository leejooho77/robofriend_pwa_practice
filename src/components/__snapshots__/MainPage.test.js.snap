// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render MainPage 1`] = `ShallowWrapper {}`;

exports[`Render MainPage Expect to render MainPage 1`] = `ShallowWrapper {}`;
